src/compiler/checker.ts

```diff
@@ -35559,7 +35547,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     hasSignatures ||= resolved.callSignatures.length !== 0 || resolved.constructSignatures.length !== 0;
                     hasApplicableSignature ||= callSignatures.length !== 0 || constructSignatures.length !== 0;
                     if (callSignatures !== resolved.callSignatures || constructSignatures !== resolved.constructSignatures) {
-                        const result = createAnonymousType(/*symbol*/ undefined, resolved.members, callSignatures, constructSignatures, resolved.indexInfos) as ResolvedType & InstantiationExpressionType;
+                        const result = createAnonymousType(createSymbol(SymbolFlags.None, InternalSymbolName.InstantiationExpression), resolved.members, callSignatures, constructSignatures, resolved.indexInfos) as ResolvedType & InstantiationExpressionType;

```

Comment:
That seems like a correct change to re-establish symbol invariants we had before instantiation expressions, but it makes me wonder: when does a type have or not have a symbol? and when should it have a symbol?

Reply:
Honestly, I'm not totally sure. It seems like many types don't have symbols; I tried adding the assert to createAnonymousType and it broke everything.

Reply:
One thing I'm wondering here is if there needs to be some sort of cache here to make sure we always use the same symbol for identical `exprType` + `type` combos. But obviously the code here works...

Reply:
I haven't looked at this PR too closely, but wondering why you aren't just using the symbol already associated with `type` here? I mean, the anonymous type being created is just a specialization of that type, but it originates in the same declaration (which is what the symbol represents).

Reply:
Good question; let me try and see what happens. I swear I tried that but I have forgotten the result.

Reply:
That does seem to work (as in, fixing the crash), though it does create some strange baseline differences (forgive the screenshot):

![image](https://github.com/microsoft/TypeScript/assets/5341706/613e0474-ff19-49b6-bc06-7d0c3af69bdd)

Now the instantiation of the `typeof ErrImpl` appears to disappear.

And stuff like:

![image](https://github.com/microsoft/TypeScript/assets/5341706/6855c582-e645-4936-9d52-4fe3540b8014)

Where `<T>` is dropped.